sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -452,14 +452,6 @@ object FoldablePropagation extends Rule[LogicalPlan] {
         case u: UnaryNode if !stop && canPropagateFoldables(u) =>
           u.transformExpressions(replaceFoldable)
 
-        // Allow inner joins. We do not allow outer join, although its output attributes are
-        // derived from its children, they are actually different attributes: the output of outer
-        // join is not always picked from its children, but can also be null.
-        // TODO(cloud-fan): It seems more reasonable to use new attributes as the output attributes
-        // of outer join.
-        case j @ Join(_, _, Inner, _) =>
```

Comment:
We forgot to check stop here. Can you just change this line into: `case j @ Join(_, _, Inner, _) if !stop =>`